packages/ember-routing/tests/system/controller_for_test.js

```diff
@@ -1,112 +1,59 @@
-import { buildOwner } from 'internal-test-helpers';
 import { EMBER_NO_DOUBLE_EXTEND } from 'ember/features';
+import { moduleFor, ApplicationTestCase } from 'internal-test-helpers';
 
-function buildInstance(namespace) {
-  let owner = buildOwner();
+moduleFor('Ember.controllerFor', class extends ApplicationTestCase {
+  ['@test controllerFor should lookup for registered controllers'](assert) {
+    this.add('controller:app', Controller.extend());
 
-  owner.__registry__.resolver = resolverFor(namespace);
-  owner.registerOptionsForType('view', { singleton: false });
-
-  owner.register('application:main', namespace, { instantiate: false });
-
-  owner.register('controller:basic', Controller, { instantiate: false });
-
-  return owner;
-}
-
-function resolverFor(namespace) {
-  return {
-    resolve(fullName) {
-      let nameParts = fullName.split(':');
-      let type = nameParts[0];
-      let name = nameParts[1];
-
-      if (name === 'basic') {
-        name = '';
-      }
-      let className = StringUtils.classify(name) + StringUtils.classify(type);
-      let factory = get(namespace, className);
-
-      if (factory) { return factory; }
-    }
-  };
-}
-
-let appInstance, appController, namespace;
-
-QUnit.module('Ember.controllerFor', {
-  setup() {
-    namespace = Namespace.create();
-    appInstance = buildInstance(namespace);
-    appInstance.register('controller:app', Controller.extend());
-    appController = appInstance.lookup('controller:app');
-  },
-  teardown() {
-    run(() => {
-      appInstance.destroy();
-      namespace.destroy();
+    return this.visit('/').then(() => {
+      let appInstance = this.applicationInstance;
+      let appController = appInstance.lookup('controller:app');
+      let controller = controllerFor(appInstance, 'app');
+      assert.equal(appController, controller, 'should find app controller');
     });
   }
 });
 
-QUnit.test('controllerFor should lookup for registered controllers', function() {
-  let controller = controllerFor(appInstance, 'app');
-
-  equal(appController, controller, 'should find app controller');
-});
-
-QUnit.module('Ember.generateController', {
-  setup() {
-    namespace = Namespace.create();
-    appInstance = buildInstance(namespace);
-  },
-  teardown() {
-    run(() => {
-      appInstance.destroy();
-      namespace.destroy();
+moduleFor('Ember.generateController', class extends ApplicationTestCase {
+  ['@test generateController should return Ember.Controller'](assert) {
+    return this.visit('/').then(() => {
+      let controller = generateController(this.applicationInstance, 'home');
+      assert.ok(controller instanceof Controller, 'should return controller');
     });
   }
-});
 
-QUnit.test('generateController should return Ember.Controller', function() {
-  let controller = generateController(appInstance, 'home');
+  ['@test generateController should return App.Controller if provided'](assert) {
```

Comment:
Can this test be changed to `should return controller:basic if resolved` instead? The global `App.Controller` has no relevance to the remaining test (and is less informative).